packages/flutter/lib/src/widgets/modal_barrier.dart

```diff
@@ -103,6 +104,8 @@ class ModalBarrier extends StatelessWidget {
           onDismiss: () {
             if (dismissible)
               Navigator.maybePop(context);
+            else
+              SystemSound.play(SystemSoundType.alert);
```

Comment:
I believe Android/iOS are not supposed to play a sound when you accidentally tap the undismissable barrier. Does this change that?

Reply:
No, this doesn't change that, because Android/iOS don't have an API to play a system sound, so on those platforms, this platform channel message will be a no-op.

Reply:
Okay, let's add some documentation to `SystemSoundType.alert` that describes the behavior on the platforms to make this clear.